src/pybind/mgr/cephadm/services/iscsi.py

```diff
@@ -142,3 +147,66 @@ def ok_to_stop(self,
         names = [f'{self.TYPE}.{d_id}' for d_id in daemon_ids]
         warn_message = f'It is presumed safe to stop {names}'
         return HandleCommandResult(0, warn_message, '')
+
+    def purge(self, service_name: str) -> None:
+        """Removes configuration
+        """
+        spec = cast(IscsiServiceSpec, self.mgr.spec_store[service_name].spec)
+        try:
+            # remove service configuration from the pool
+            rm_pool_thread = Thread(target=remove_pool, args=(self.mgr, spec.pool))
```

Comment:
I think we need to shell out. just a thread might not help us.

Reply:
How do you suggest to shell out? multiprocessing module will add another mgr instance into the container and can produce problems.

Reply:
i think that to "shell out" the rados command only can be achieved through "_run_cephadm" ... ( or making possible in the mgr container to use ceph commands). if we try to execute the "rados" command directly in the mgr container we have this:

```
[ceph: root@cephlab2-node-00 /]# rados  -p myscsi ls
2021-04-12T14:45:42.483+0000 7f90c006cd00 -1 auth: unable to find a keyring on /etc/ceph/ceph.client.admin.keyring,/etc/ceph/ceph.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin,: (2) No such file or directory
2021-04-12T14:45:42.483+0000 7f90c006cd00 -1 AuthRegistry(0x55fd4904f440) no keyring found at /etc/ceph/ceph.client.admin.keyring,/etc/ceph/ceph.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin,, disabling cephx
2021-04-12T14:45:42.484+0000 7f90c006cd00 -1 auth: unable to find a keyring on /etc/ceph/ceph.client.admin.keyring,/etc/ceph/ceph.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin,: (2) No such file or directory
2021-04-12T14:45:42.484+0000 7f90c006cd00 -1 AuthRegistry(0x7ffd0e55ad00) no keyring found at /etc/ceph/ceph.client.admin.keyring,/etc/ceph/ceph.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin,, disabling cephx
2021-04-12T14:45:42.485+0000 7f90b821d700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [1]
2021-04-12T14:45:42.486+0000 7f90b8a1e700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [1]
2021-04-12T14:45:42.486+0000 7f90b921f700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [1]
2021-04-12T14:45:42.486+0000 7f90c006cd00 -1 monclient: authenticate NOTE: no keyring found; disabled cephx authentication
failed to fetch mon config (--no-mon-config to skip)
 
 
 
[ceph: root@cephlab2-node-00 /]# rados -c /etc/ceph/ceph.conf -p myscsi ls  
2021-04-12T14:47:23.129+0000 7f5b8cd5bd00 -1 auth: unable to find a keyring on /etc/ceph/ceph.client.admin.keyring,/etc/ceph/ceph.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin,: (2) No such file or directory
2021-04-12T14:47:23.129+0000 7f5b8cd5bd00 -1 AuthRegistry(0x562c24dfb440) no keyring found at /etc/ceph/ceph.client.admin.keyring,/etc/ceph/ceph.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin,, disabling cephx
2021-04-12T14:47:23.130+0000 7f5b8cd5bd00 -1 auth: unable to find a keyring on /etc/ceph/ceph.client.admin.keyring,/etc/ceph/ceph.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin,: (2) No such file or directory
2021-04-12T14:47:23.130+0000 7f5b8cd5bd00 -1 AuthRegistry(0x7fff123a85e0) no keyring found at /etc/ceph/ceph.client.admin.keyring,/etc/ceph/ceph.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin,, disabling cephx
2021-04-12T14:47:23.130+0000 7f5b85f0e700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [1]
2021-04-12T14:47:23.131+0000 7f5b84f0c700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [1]
2021-04-12T14:47:23.131+0000 7f5b8570d700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [1]
2021-04-12T14:47:23.131+0000 7f5b8cd5bd00 -1 monclient: authenticate NOTE: no keyring found; disabled cephx authentication
failed to fetch mon config (--no-mon-config to skip) 
```

Reply:
What is the preferred implementation? 
_run_cephadm 
or
 make mgr container able to execute ceph/rados commands?

Reply:
```
-            cmd = ['radosgw-admin',
-                   '--key=%s' % keyring,
-                   '--user', 'rgw.%s' % rgw_id,
-                   'realm', 'list',
-                   '--format=json']
-            result = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            out = result.stdout
-            if not out:
-                return []
-            try:
-                j = json.loads(out)
-                return j.get('realms', [])
-            except Exception:
-                raise OrchestratorError('failed to parse realm info')
```
for the keyring, pass the keyring file for the mgr itself, which should be ``rados -k /var/lib/ceph/mgr/ceph-{{mgr_id}}/keyring -i {{mgr_id}} ..``